homeassistant/components/knx/project.py

```diff
@@ -80,23 +81,33 @@ def initial_state(self) -> None:
         self.group_addresses = {}
         self.info = None
 
-    async def load_project(self, data: KNXProjectModel | None = None) -> None:
+    async def load_project(
+        self, xknx: XKNX, data: KNXProjectModel | None = None
+    ) -> None:
         """Load project data from storage."""
         if project := data or await self._store.async_load():
             self.devices = project["devices"]
             self.info = project["info"]
+            xknx.group_address_dpt.clear()
+            xknx_ga_dict = {}
 
             for ga_model in project["group_addresses"].values():
                 ga_info = _create_group_address_info(ga_model)
                 self.group_addresses[ga_info.address] = ga_info
+                if (dpt_model := ga_model.get("dpt")) is not None:
+                    xknx_ga_dict[ga_model["address"]] = dpt_model
+
+            xknx.group_address_dpt.set(xknx_ga_dict)  # type: ignore[arg-type]
 
             _LOGGER.debug(
                 "Loaded KNX project data with %s group addresses from storage",
                 len(self.group_addresses),
             )
             self.loaded = True
 
-    async def process_project_file(self, file_id: str, password: str) -> None:
+    async def process_project_file(
+        self, file_id: str, password: str, xknx: XKNX
+    ) -> None:
```

Comment:
Personal preference thing, I always like my bigger objects first in the parameter list, like we always have `hass: HomeAssistant` first

Reply:
👍